tests/main/preseed-reset/task.yaml

```diff
@@ -20,6 +20,21 @@ prepare: |
   . "$TESTSLIB/preseed.sh"
   mount_ubuntu_image "$(pwd)/cloudimg.img" "$IMAGE_MOUNTPOINT"
 
+  #shellcheck source=tests/lib/snaps.sh
+  . "$TESTSLIB/snaps.sh"
+  # add snapd from this branch into the seed
+  repack_snapd_deb_into_snapd_snap "$PWD"
+  mv snapd-from-deb.snap snapd.snap
+  inject_snap_into_seed "$IMAGE_MOUNTPOINT" snapd
+
+  # Update the snapd deb to the version under test.
+  cp "${GOHOME}"/snapd_*.deb "$IMAGE_MOUNTPOINT/tmp/snapd.deb"
+  chroot "$IMAGE_MOUNTPOINT" dpkg -i /tmp/snapd.deb
```

Comment:
this spread test installs the snapd deb in the image mountpoint, but the other test we have (tests/main/preseed) just injects the new snapd snap into the seed, but doesn't install the new snapd deb in the image. 
I feel like we should ideally be testing both variants, where we both test that a new snapd from the branch works as both the snapd snap in the seed, and the snapd deb in the image (i.e. all the stars are aligned and the same version is being used everywhere) and also when we have different versions, i.e. the version of snapd deb is _newer_ than the version in the snapd snap in the seed, and the version of snapd deb is older than the version in the snapd snap in the seed. I'm not sure we have full coverage of that at the moment and there have been related bugs to this that I believe all have unit tests now, but spread tests are nice to have as well.

Reply:
That's a valid observation and good idea about testing variants in preseeding tests (I'm going to prepare a PR for that), *except for this test* I think. The main point of this test is checking that `/usr/lib/snapd/snap-preseed --reset ...` removes any snapd artifacts, and this is always running the current code.

The problem with this test is that that snapd deb may occasionally introduce new (empty) directories (e.g. /var/lib/snapd/something) at the packaging level, and even though we run the right version of snapd during preseeding, the state of the filesystem that we record in `before-preseeding.txt` may not agree with `after-reset.txt` state, which annoys the diff check (I couldn't find any better way of testing if reset really cleans filesystem up when I originally implemented this), and this is what this PR tries to solve.

I hope this makes sense?

Reply:
do we have a real bug though when snap-preseed --reset is used in practice because the deb of the host builder is older than the snapd snap in the system?  what I mean, and I don't remember, should --reset also trampoline into the snapd snap if it's newer?

Reply:
Reset doesn't trampoline into snapd snap at the moment, the idea was that snap-preseed is always the latest and "knows" all the directories that ever existed in previous snapd versions. If we don't want to / can't assume that, then yes we would need to trampoline.

Reply:
yea, I'm not sure that the assumption is strongly valid though? trampoling is a bit of code but maybe not a lot?  I would try to do that instead but without rush